rust/agama-lib/src/storage/client/zfcp.rs

```diff
@@ -153,25 +162,40 @@ impl<'a> ZFCPClient<'a> {
     ) -> Result<Vec<(OwnedObjectPath, ZFCPController)>, ServiceError> {
         let managed_objects = self.object_manager_proxy.get_managed_objects().await?;
 
-        let mut devices: Vec<(OwnedObjectPath, ZFCPController)> = vec![];
+        let mut devices: Vec<(OwnedObjectPath, PartialZFCPController)> = vec![];
         for (path, ifaces) in managed_objects {
             if let Some(properties) = ifaces.get("org.opensuse.Agama.Storage1.ZFCP.Controller") {
                 let id = extract_id_from_path(&path)?.to_string();
                 let channel: String = get_property(properties, "Channel")?;
-                let luns_map = get_luns_map(id.as_str()).await?;
                 devices.push((
                     path,
-                    ZFCPController {
+                    PartialZFCPController {
                         id: id,
                         channel: channel,
                         lun_scan: get_property(properties, "LUNScan")?,
                         active: get_property(properties, "Active")?,
-                        luns_map: luns_map, // try to use optimized thred version here
                     },
                 ))
             }
         }
-        Ok(devices)
+        let mut tasks = vec![];
+        for (_path, partial_controller) in &devices {
+            tasks.push(
+                tokio::spawn(get_luns_map(partial_controller.channel.clone()))
```

Comment:
with this additional paralellization time is reduced to 5.3 seconds. So basically for two controllers with 3 LUNS in total it is gets to half of time. I expect that for bigger systems it will be even more significant.